portfolio/src/main/java/com/google/sps/servlets/UserInfoServlet.java

```diff
@@ -36,28 +36,29 @@
 @WebServlet("/user-info")
 public class UserInfoServlet extends HttpServlet {
   private UserDaoDatastore userStorage = new UserDaoDatastore();
+  private Gson gson = new Gson();
 
   @Override
   public void init() {
     UserDaoDatastore userStorage = new UserDaoDatastore();
   }
 
-  private String toJson(String status) {
-    Gson gson = new Gson();
-    return gson.toJson(status);
-  }
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("application/json");
     UserService userService = UserServiceFactory.getUserService();
     if (userService.isUserLoggedIn()) {
-      String nickname = getUserNickname(userService.getCurrentUser().getEmail());
-      response.getWriter().println(toJson(nickname));
+        String email = userService.getCurrentUser().getEmail(); // user email is used as id
+        Optional<User> user = userStorage.get(email);
+        if (user.isPresent()) {
+            response.getWriter().println(gson.toJson(user.get()));
+        } else {
+            response.getWriter().println(gson.toJson(user.empty()));
+        }
 
     } else {
       String loginUrl = userService.createLoginURL("/auth");
-      response.getWriter().println(toJson(loginUrl));
+      response.getWriter().println(gson.toJson(loginUrl));
```

Comment:
Is this URL relevant anymore? Don't we have the login/logout URLs in our user-auth.template.html?

Reply:
huh, it's not super relevant anymore, I just wanted to return something that would indicate that the user wasn't logged in, although I suppose I could just return a string saying "Logged Out" or something

Reply:
Yeah, that would make the javascript function for determining if someone is logged out much cleaner too!